core/src/include/Xios.hpp

```diff
@@ -30,12 +30,37 @@ namespace Nextsim {
 void enableXios();
 
 class Xios : public Configured<Xios> {
-public:
+private:
+    static Xios* instancePtr;
+
+    // Private constructor
     Xios(const std::string dt = "P0-0T01:00:00", const std::string contextid = "nextSIM-DG",
         const std::string starttime = "1970-01-01T00:00:00Z",
         const std::string calendartype = "Gregorian");
+
+public:
+    Xios(const Xios& xiosHandler) = delete;
     ~Xios();
 
+    /*
+     * Define Xios handler Singleton
+     *
+     * NOTE: The arguments will only be used the first time this is called.
+     */
+    inline static Xios* getInstance(const std::string dt = "P0-0T01:00:00",
+        const std::string contextId = "nextSIM-DG",
+        const std::string startTime = "1970-01-01T00:00:00Z",
+        const std::string calendarType = "Gregorian")
+    {
+        if (isNull()) {
+            instancePtr = new Xios(dt, contextId, startTime, calendarType);
```

Comment:
If the `Xios` class was a static function variable, then it wouldn't have to be dynamically allocated.

Reply:
Again, see 10d6207d42813c5009e7a69bba47c07bfd05f110.